src/dsp_tools/commands/ingest_xmlupload/upload_xml.py

```diff
@@ -32,28 +35,37 @@ def ingest_xmlupload(
         creds: credentials to access the DSP server
         interrupt_after: if set, the upload will be interrupted after this number of resources
 
+    Returns:
+        True if all resources could be uploaded without errors; False if one of the resources could not be
+        uploaded because there is an error in it
+
     Raises:
         InputError: if any media was not uploaded or uploaded media was not referenced.
     """
-    xml_tree_orig = etree.parse(xml_file)
-    xml_tree_orig = remove_comments_from_element_tree(xml_tree_orig)
-
-    shortcode = xml_tree_orig.getroot().attrib["shortcode"]
-    orig_path_2_id_filename = get_mapping_dict_from_file(shortcode)
-    xml_tree_replaced, ingest_info = replace_filepath_with_internal_filename(
-        xml_tree=xml_tree_orig,
-        orig_path_2_id_filename=orig_path_2_id_filename,
-    )
-    if ok := ingest_info.ok_msg():
-        print(ok)
-        logger.info(ok)
-    else:
-        err_msg = ingest_info.execute_error_protocol()
-        raise InputError(err_msg)
-
-    xmlupload(
-        input_file=xml_tree_replaced,
-        creds=creds,
-        imgdir=".",
-        config=UploadConfig(media_previously_uploaded=True, interrupt_after=interrupt_after),
+    default_ontology, root, shortcode = validate_and_parse_xml_file_preprocessing_done(xml_file)
```

Comment:
I would like to rename both `validate_and_parse_xml_file` and `validate_and_parse_xml_file_preprocessing_done` (and the file containing it) because to me it seems that on the high level, it doesn't matter what exactly happens in there. The real concern is that given an xmlupload XML file, they turn it into some information for the following steps. My best idea right now is `parse_xmlupload_file` and `parse_ingest_xmlupload_file`, reusing the names of the CLI commands they correspond to. What do you think?

Reply:
I agree that "validate" can be omitted. But I don't like "ingest_xmlupload_file", because we only have 1 XML file format. So my counter-suggestion is: 

- `parse_xml_file()`
- `parse_xml_file_if_preprocessing_done()`

Reply:
technically we have only one XML file format; but at the same time, they have different meaning in the different contexts of `xmlupload` and `ingest-xmlupload` and need to be handled differently. In that sense, they are not really the same. And I have very strong feelings about using the word "if" in function names. 😉   
How would you feel about 

- `parse_xml_for_xmlupload()`
- `parse_xml_for_ingest_xmlupload()` ?

Now that I write this, I would actually even want to move these functions to the `xmlupload.py` and `upload_xml.py` files.  
And then the whole discussion is obsolete, because both could be private functions in their respective file called `_parse_xml()`. WDYT?

Reply:
Moving would be the best solution, this would solve the problem.